### DISCUSSION TITLE
OpenID does not work for microsoft as identity provider

### DISCUSSION THREAD

**Discussion by erktm** [photoprism/photoprism#4451] [Q&A]
Hi there,

I am pretty sure this should be a quick fix. I am not able to connect or redirect to Microsoft via PhotoPrism with openID. 

![image](https://github.com/user-attachments/assets/003b4262-c62f-4d4a-ab74-b8265b1efcc0)

My config looks like this:
`OPENID CONNECT (OIDC)

|---------------|----------------------------------------------------------------|
|     Name      |                             Value                              |
|---------------|----------------------------------------------------------------|
| oidc-uri      | https://login.microsoftonline.com/common/oauth2/v2.0/authorize |
| oidc-client   | XXXXXXXXXXXXXXXXXXXXXXXXXXX                          |
| oidc-secret   | ****************************************                       |
| oidc-scopes   | openid email profile address                                   |
| oidc-provider | Microsoft                                                      |
| oidc-icon     | /static/img/oidc.svg                                           |
| oidc-redirect | true                                                           |
| oidc-register | true                                                           |
| oidc-username | name                                                           |
| oidc-role     | guest                                                          |
| oidc-webdav   | false                                                          |
| disable-oidc  | false                                                          |
|---------------|----------------------------------------------------------------|`

Also there is no registration button to create an account. Please let me know what am I doing wrong.

Thanks
Erik

**Comment by graciousgrey** [comment]
Hello,

You might get some insights on what might be the problem by turning on trace logs and watching the logs while trying to authenticate with OIDC. 

https://docs.photoprism.app/getting-started/troubleshooting/logs/#__tabbed_1_3

As for the registration there is no separate button. If registration is turned on, a new user is created in PhotoPrism when a oidc user signs in for the first time. In case registration is turned off the admin needs to create the user first.

https://docs.photoprism.app/developer-guide/api/oidc/

https://docs.photoprism.app/user-guide/users/cli/#command-options
